core/agent.py

```
@@ -1,172 +1,250 @@
+                after = line[len("tool:"):].strip()
+                name, rest = after.split("(", 1)
+                name = name.strip()
+                
+                if not rest.endswith(")"):
+                    continue
+                    
+                json_str = rest[:-1].strip()
+                args = json.loads(json_str)
+                invocations.append((name, args))
+            except Exception:
+                continue
```

Ruff (0.15.10)

[error] `try`-`except`-`continue` detected, consider logging the exception

(S112)